src/Chapter3.hs

```diff
@@ -971,6 +1056,25 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Day = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday deriving (Enum)
+
+isWeekend :: Day -> Bool
+isWeekend Sunday   = True
+isWeekend Saturday = True
+isWeekend _        = False
+
+-- maybe there's a better way to make `succ` wrap here??
+nextDay :: Day -> Day
+nextDay Sunday = Monday
+nextDay day    = succ day
+
+daysToParty :: Day -> Int
+daysToParty day = go day 0
+ where
+  go :: Day -> Int -> Int
+  go Friday n = n
+  go day'   n = go (nextDay day') (n + 1)
```

Comment:
Nice recursive implementation! But you can implement it more efficiently by using the `fromEnum` method of the `Enum` typeclass